osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -112,7 +111,17 @@ public void Import(params string[] paths)
                     }
                 }
 
-                Import(new[] { beatmapSet });
+                try
+                {
+                    Import(beatmapSet);
+                }
+                catch
+                {
+                }
+                finally
+                {
+                    File.Delete(p);

```

Comment:
- imported maps should only be deleted on success.
- the file.delete should also be handled here rather than left to throw exceptions.

ie. the two things i already commented on above.